### PR TITLE
Nodejs week1

### PR DESCRIPTION
**Pull request by andrii1**

### CODE REVIEW

File: src/backend/app.js [hunk 1]
```diff
@@ -21,10 +31,50 @@ app.use(cors());
...
 
+//testing knex connection
+router.get("/info", async (req, res) => {
+  const dbResult = await knex.raw("SELECT VERSION()");
+  const row = dbResult[0][0];
+  res.json({ nodeVersion: process.version, mysqlVersion: row["VERSION()"] });
+});
+
+router.get("/future-meals", async (req, res) => {
+  const [rows] = await knex.raw("SELECT * from meal WHERE `when` > CURDATE()");
```
Comment (Divyajg) [inline src/backend/app.js]: Where condition should be `>=` for `future meals`

Comment (Divyajg) [reply src/backend/app.js]: It is good idea to give descriptive names to understand the purpose of that variable.

Comment (Divyajg) [reply src/backend/app.js]: Not suggested to select `*`, instead select whatever specifically you want from the table like, `title`/`price`/etc.

File: src/backend/app.js [hunk 2]
```diff
@@ -21,10 +31,50 @@ app.use(cors());
...
+  res.json({ rows });
+});
+
+router.get("/first-meal", async (req, res) => {
+  const [rows] = await knex.raw("SELECT * from meal ORDER by id Asc LIMIT 1");
+  if (rows.length === 0) {
+    res.status(404).send("There are no meals...");
+  } else {
+    res.json({ rows });
+  }
```
Comment (Divyajg) [inline src/backend/app.js:80]: You can use this `if/else` statement for all the requests.

**Review by Divyajg — COMMENTED** [review]
Good work. 👍
I provided a few suggestions that applies to all the requests.

File: src/backend/app.js [hunk 3]
```diff
@@ -38,6 +38,19 @@ router.get("/info", async (req, res) => {
...
 });
+
+router.get("/future-meals", async (req, res) => {
+  const currentTime = Date.now();
+  const [rows] = await knex.raw("SELECT * from meal WHERE `when` > CURDATE()");
+  res.json({ rows });
+});
+
+router.get("/past-meals", async (req, res) => {
+  const currentTime = Date.now();
```
Comment (27ManishaPatel) [inline src/backend/app.js]: where 'const currentTime' variable  are you using ?  as CURDATE()  returns the current date right !

File: src/backend/app.js [hunk 4]
```diff
@@ -30,27 +30,47 @@ app.use(express.json());
...
+  if (rows.length === 0) {
+    res.status(404).send("There are no meals...");
+  } else {
+    res.json({ rows });
+  }
+});
+
 if (process.env.API_PATH) {
   app.use(process.env.API_PATH, router);
 } else {
```
Comment (27ManishaPatel) [inline src/backend/app.js:96]: Nice !

**Review by 27ManishaPatel — APPROVED** [review]
Great with all the queries

File: src/backend/app.js (hunk 1 above)
Comment (andrii1) [reply src/backend/app.js]: > Where condition should be `>=` for `future meals`

changed

Comment (andrii1) [reply src/backend/app.js]: > Not suggested to select `*`, instead select whatever specifically you want from the table like, `title`/`price`/etc.

changed to title

File: src/backend/app.js (hunk 2 above)
Comment (andrii1) [reply src/backend/app.js:80]: changed

**Pull request merged**